client/src/partials/caution/caution.js

```diff
@@ -251,6 +251,35 @@ angular.module('bhima.controllers')
       $scope.setCurrency(sessionDefault);
     }
 
+    function haltOnNoExchange () {
+      if (exchange.hasDailyRate()) { return; }
+
+      var instance = $modal.open({
+        templateUrl : 'noExchangeRate.html',
+        backdrop    : 'static',
+        keyboard    : false,
+        controller  : function ($scope, $modalInstance) {
+          $scope.timestamp= new Date();
+
+          $scope.close = function close () {
+            $modalInstance.dismiss();
+          };
+
+          $scope.setExchange = function setExchange () {
```

Comment:
`$scope.setExchange()` should probably redirect the user to `#/exchange_rate` using `$location`.